Jasmine/Common/Models/Coordinate.swift

```diff
@@ -18,6 +18,19 @@ struct Coordinate {
         self.row = row
         self.col = col
     }
+
+    /// Compares a coordinate with another coordinate. Returns true if and only if self < other.
+    /// Here comparison is done by looking at the row, then by column.
+    ///
+    /// - Parameter other: the other coordinate to be compared
+    /// - Returns: true if and only if self < other
+    func isLessThanByRowFirst(_ other: Coordinate) -> Bool {
```

Comment:
by definition of coordinate, numbering starts from top left and goes rightwards, then downwards. So maybe should implement comparable (<) (swift equivalent) instead? what do you think?